### PR TITLE
Match check_for_interactables

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

File: src/77480.c [hunk 1]
```diff
@@ -603,7 +605,111 @@ void func_800E06C0(s32 arg0) {
...
+                phi_s2 = 0;
+                if ((phi_s0 & 0x2000) == 0) {
+                    if (should_collider_allow_interact(phi_s0) == FALSE) {
+                        playerStatus->unk_C6 = -1;
+                        playerStatus->flags &= ~PLAYER_STATUS_FLAGS_8000000;
+                        return;
+                    }
+                }
+            } else {
+                if (phys_can_player_interact() == FALSE) {
```
Comment (nanaian) [inline src/77480.c:671]:
```suggestion
                if (!phys_can_player_interact()) {
```

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.02%) on `us`.

✅ This PR fixes 1 warnings!

File: src/77480.c [hunk 2]
```diff
@@ -603,7 +605,115 @@ void func_800E06C0(s32 arg0) {
 
 INCLUDE_ASM(s32, "77480", func_800E06D8);
 
-INCLUDE_ASM(s32, "77480", check_for_interactables);
+void check_for_interactables(void) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Npc* npc = gPlayerStatus.unk_C8;
+    s32 phi_s2;
+
+    if (playerStatus->animFlags & PLAYER_STATUS_ANIM_FLAGS_100 || D_8010C940 || D_8010C920) {
```
Comment (ethteck) [inline src/77480.c:613]: I know this goes against what we normally do, but I'd prefer to wrap the flag check part of this in parens to make it clearer next to those other bool checks

File: src/77480.c [hunk 3]
```diff
@@ -603,7 +605,115 @@ void func_800E06C0(s32 arg0) {
...
+                    return;
+                }
+                phi_s2 = 0;
+            } else {
+                playerStatus->unk_C6 = -1;
+                playerStatus->flags &= ~PLAYER_STATUS_FLAGS_8000000;
+                return;
+            }
+        } else {
+            if ((phi_s0 & 0x4000) == 0) {
```
Comment (ethteck) [inline src/77480.c:661]: ! for flag checks (also below)

File: src/77480.c [hunk 4]
```diff
@@ -603,7 +605,115 @@ void func_800E06C0(s32 arg0) {
...
+                phi_s2 = 0;
+                if ((phi_s0 & 0x2000) == 0) {
+                    if (should_collider_allow_interact(phi_s0) == FALSE) {
```
Comment (ethteck) [inline src/77480.c:664]: ! for bool check

File: src/77480.c [hunk 5]
```diff
@@ -603,7 +605,115 @@ void func_800E06C0(s32 arg0) {
...
+            if ((playerStatus->flags & PLAYER_STATUS_FLAGS_8000000)) {
+                return;
+            }
+        } else {
+            playerStatus->flags &= ~PLAYER_STATUS_FLAGS_8000000;
+        }
+
+        playerStatus->unk_C6 = phi_s0;
+        if ((phi_s2 == 0) || phi_s0 >= 0 && get_entity_by_index(phi_s0)->flags & ENTITY_FLAGS_SHOWS_INSPECT_PROMPT) {
+            if ((u8) playerStatus->actionState < 3) {
```
Comment (ethteck) [inline src/77480.c:690]: playerStatus->actionState == IDLE || playerStatus->actionState == WALK || playerStatus->actionState == RUN (I think these start with ACTION_STATE_)

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/77480.c (hunk 5 above)
Comment (JoshDuMan) [reply src/77480.c:690]: I had no idea this worked

**Pull request merged**